File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/auth/repository/GroupRepository.java
package uk.gov.justice.digital.hmpps.oauth2server.auth.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.lang.NonNull;
import uk.gov.justice.digital.hmpps.oauth2server.auth.model.Group;

import java.util.List;
import java.util.Optional;

public interface GroupRepository extends CrudRepository<Group, String> {
    @NonNull
    List<Group> findAllByOrderByGroupName();

    Optional<Group> findByGroupCode(String groupCode);
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/auth/repository/GroupRepository.kt
package uk.gov.justice.digital.hmpps.oauth2server.auth.repository

import org.springframework.data.repository.CrudRepository
import uk.gov.justice.digital.hmpps.oauth2server.auth.model.Group
import java.util.Optional

interface GroupRepository : CrudRepository<Group, String> {

  fun findAllByOrderByGroupName(): List<Group>
  fun findByGroupCode(groupCode: String?): Optional<Group>
}
